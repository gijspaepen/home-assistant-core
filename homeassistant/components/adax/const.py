"""Constants for the Adax integration."""

from typing import Final

from homeassistant.const import Platform

ACCOUNT_ID: Final = "account_id"
CLOUD = "Cloud"
CONNECTION_TYPE = "connection_type"
DOMAIN: Final = "adax"
LOCAL = "Local"
WIFI_SSID = "wifi_ssid"
WIFI_PSWD = "wifi_pswd"

PLATFORMS = [Platform.CLIMATE, Platform.SENSOR]
