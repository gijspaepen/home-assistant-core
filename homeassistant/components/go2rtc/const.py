"""Go2rtc constants."""

DOMAIN = "go2rtc"

CONF_DEBUG_UI = "debug_ui"
EXCLUSE_ERROR_MESSAGE = "Url and debug_ui cannot be set at the same time."
