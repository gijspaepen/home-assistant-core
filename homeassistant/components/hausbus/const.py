"""Constants for the Haus-Bus integration."""

from __future__ import annotations

import logging

LOGGER = logging.getLogger(__package__)

DOMAIN = "hausbus"
ATTR_ON_STATE = "on_state"
