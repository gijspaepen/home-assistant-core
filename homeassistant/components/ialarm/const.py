"""Constants for the iAlarm integration."""

from pyialarm import IAlarm

from homeassistant.components.alarm_control_panel import AlarmControlPanelEntityState

DATA_COORDINATOR = "ialarm"

DEFAULT_PORT = 18034

DOMAIN = "ialarm"

IALARM_TO_HASS = {
    IAlarm.ARMED_AWAY: AlarmControlPanelEntityState.ARMED_AWAY,
    IAlarm.ARMED_STAY: AlarmControlPanelEntityState.ARMED_HOME,
    IAlarm.DISARMED: AlarmControlPanelEntityState.DISARMED,
    IAlarm.TRIGGERED: AlarmControlPanelEntityState.TRIGGERED,
}
