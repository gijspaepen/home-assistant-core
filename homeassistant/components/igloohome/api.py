class Api:  # noqa: D100, D101
    def __init__(self) -> None:  # noqa: D107
        pass
