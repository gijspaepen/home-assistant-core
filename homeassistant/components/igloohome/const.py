"""Constants for the igloohome integration."""

DOMAIN = "igloohome"

OAUTH2_AUTHORIZE = "https://auth.igloohome.co/login"
OAUTH2_TOKEN = "https://auth.igloohome.co/oauth2/token"
