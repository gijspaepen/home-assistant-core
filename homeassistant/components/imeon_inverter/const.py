"""Constant for Imeon component."""

DOMAIN = "imeon_inverter"
HUBNAME = "imeon_inverter_hub"
TIMEOUT = 10  # seconds
