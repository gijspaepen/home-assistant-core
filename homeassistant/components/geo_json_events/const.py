"""Define constants for the GeoJSON events integration."""

from __future__ import annotations

from typing import Final

from homeassistant.const import Platform

DOMAIN: Final = "geo_json_events"

PLATFORMS: Final = [Platform.GEO_LOCATION]

ATTR_EXTERNAL_ID: Final = "external_id"
DEFAULT_RADIUS_IN_KM: Final = 20.0
DEFAULT_RADIUS_IN_M: Final = 20000.0
DEFAULT_UPDATE_INTERVAL: Final = 300
SOURCE: Final = "geo_json_events"
CONF_UPDATE_INTERVAL: Final = "update_interval"

SIGNAL_DELETE_ENTITY: Final = "geo_json_events_delete_{}"
SIGNAL_UPDATE_ENTITY: Final = "geo_json_events_update_{}"
