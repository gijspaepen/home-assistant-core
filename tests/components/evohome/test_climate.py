"""The tests for climate entities of evohome.

All evohome systems have controllers and at least one zone.
"""

from __future__ import annotations

from unittest.mock import patch

from freezegun.api import FrozenDateTimeFactory
import pytest
from syrupy import SnapshotAssertion

from homeassistant.components.climate import (
    ATTR_HVAC_MODE,
    ATTR_PRESET_MODE,
    SERVICE_SET_HVAC_MODE,
    SERVICE_SET_PRESET_MODE,
    SERVICE_SET_TEMPERATURE,
    HVACMode,
)
from homeassistant.const import ATTR_ENTITY_ID, ATTR_TEMPERATURE, Platform
from homeassistant.core import HomeAssistant

from .const import TEST_INSTALLS


@pytest.mark.parametrize("install", TEST_INSTALLS)
async def test_zone_set_hvac_mode(
    hass: HomeAssistant,
    zone_id: str,
    snapshot: SnapshotAssertion,
) -> None:
    """Test climate methods of a evohome-compatible zone."""

    results = []

    # SERVICE_SET_HVAC_MODE(HVACMode.HEAT)
    with patch("evohomeasync2.zone.Zone.reset_mode") as mock_fcn:
        await hass.services.async_call(
            Platform.CLIMATE,
            SERVICE_SET_HVAC_MODE,
            {
                ATTR_ENTITY_ID: zone_id,
                ATTR_HVAC_MODE: HVACMode.HEAT,
            },
            blocking=True,
        )

        assert mock_fcn.await_count == 1
        assert mock_fcn.await_args.args == ()
        assert mock_fcn.await_args.kwargs == {}

    # SERVICE_SET_HVAC_MODE(HVACMode.OFF)
    with patch("evohomeasync2.zone.Zone.set_temperature") as mock_fcn:
        await hass.services.async_call(
            Platform.CLIMATE,
            SERVICE_SET_HVAC_MODE,
            {
                ATTR_ENTITY_ID: zone_id,
                ATTR_HVAC_MODE: HVACMode.OFF,
            },
            blocking=True,
        )

        assert mock_fcn.await_count == 1

        results.append(mock_fcn.await_args.args)
        results.append(mock_fcn.await_args.kwargs)

    assert results == snapshot


@pytest.mark.parametrize("install", TEST_INSTALLS)
async def test_zone_set_preset_mode(
    hass: HomeAssistant,
    zone_id: str,
    freezer: FrozenDateTimeFactory,
    snapshot: SnapshotAssertion,
) -> None:
    """Test climate methods of a evohome-compatible zone."""

    freezer.move_to("2024-07-10T12:00:00Z")
    results = []

    # SERVICE_SET_PRESET_MODE(none)
    with patch("evohomeasync2.zone.Zone.reset_mode") as mock_fcn:
        await hass.services.async_call(
            Platform.CLIMATE,
            SERVICE_SET_PRESET_MODE,
            {
                ATTR_ENTITY_ID: zone_id,
                ATTR_PRESET_MODE: "none",
            },
            blocking=True,
        )

        assert mock_fcn.await_count == 1
        assert mock_fcn.await_args.args == ()
        assert mock_fcn.await_args.kwargs == {}

    # SERVICE_SET_PRESET_MODE(permanent)
    with patch("evohomeasync2.zone.Zone.set_temperature") as mock_fcn:
        await hass.services.async_call(
            Platform.CLIMATE,
            SERVICE_SET_PRESET_MODE,
            {
                ATTR_ENTITY_ID: zone_id,
                ATTR_PRESET_MODE: "permanent",
            },
            blocking=True,
        )

        assert mock_fcn.await_count == 1

        results.append(mock_fcn.await_args.args)
        results.append(mock_fcn.await_args.kwargs)

    # SERVICE_SET_PRESET_MODE(temporary)
    with patch("evohomeasync2.zone.Zone.set_temperature") as mock_fcn:
        await hass.services.async_call(
            Platform.CLIMATE,
            SERVICE_SET_PRESET_MODE,
            {
                ATTR_ENTITY_ID: zone_id,
                ATTR_PRESET_MODE: "temporary",
            },
            blocking=True,
        )

        assert mock_fcn.await_count == 1

        results.append(mock_fcn.await_args.args)
        results.append(mock_fcn.await_args.kwargs)

    assert results == snapshot


@pytest.mark.parametrize("install", TEST_INSTALLS)
async def test_zone_set_temperature(
    hass: HomeAssistant,
    zone_id: str,
    freezer: FrozenDateTimeFactory,
    snapshot: SnapshotAssertion,
) -> None:
    """Test climate methods of a evohome-compatible zone."""

    freezer.move_to("2024-07-10T12:00:00Z")
    results = []

    # SERVICE_SET_TEMPERATURE(temp)
    with patch("evohomeasync2.zone.Zone.set_temperature") as mock_fcn:
        await hass.services.async_call(
            Platform.CLIMATE,
            SERVICE_SET_TEMPERATURE,
            {
                ATTR_ENTITY_ID: zone_id,
                ATTR_TEMPERATURE: 19.1,
            },
            blocking=True,
        )

        assert mock_fcn.await_count == 1

        results.append(mock_fcn.await_args.args)
        results.append(mock_fcn.await_args.kwargs)

    assert results == snapshot
