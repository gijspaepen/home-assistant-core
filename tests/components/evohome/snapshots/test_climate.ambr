# serializer version: 1
# name: test_zone_set_hvac_mode[default]
  list([
    tuple(
      5.0,
    ),
  ])
# ---
# name: test_zone_set_hvac_mode[h032585]
  list([
    tuple(
      4.5,
    ),
  ])
# ---
# name: test_zone_set_hvac_mode[h099625]
  list([
    tuple(
      5.0,
    ),
  ])
# ---
# name: test_zone_set_hvac_mode[minimal]
  list([
    tuple(
      5.0,
    ),
  ])
# ---
# name: test_zone_set_hvac_mode[sys_004]
  list([
    tuple(
      5.0,
    ),
  ])
# ---
# name: test_zone_set_preset_mode[default]
  list([
    tuple(
      17.0,
    ),
    tuple(
      17.0,
    ),
    dict({
      'until': datetime.datetime(2024, 7, 10, 21, 10, tzinfo=datetime.timezone.utc),
    }),
  ])
# ---
# name: test_zone_set_preset_mode[h032585]
  list([
    tuple(
      21.5,
    ),
    tuple(
      21.5,
    ),
    dict({
      'until': datetime.datetime(2024, 7, 10, 21, 10, tzinfo=datetime.timezone.utc),
    }),
  ])
# ---
# name: test_zone_set_preset_mode[h099625]
  list([
    tuple(
      21.5,
    ),
    tuple(
      21.5,
    ),
    dict({
      'until': datetime.datetime(2024, 7, 10, 19, 10, tzinfo=datetime.timezone.utc),
    }),
  ])
# ---
# name: test_zone_set_preset_mode[minimal]
  list([
    tuple(
      17.0,
    ),
    tuple(
      17.0,
    ),
    dict({
      'until': datetime.datetime(2024, 7, 10, 21, 10, tzinfo=datetime.timezone.utc),
    }),
  ])
# ---
# name: test_zone_set_preset_mode[sys_004]
  list([
    tuple(
      15.0,
    ),
    tuple(
      15.0,
    ),
    dict({
      'until': datetime.datetime(2024, 7, 10, 20, 10, tzinfo=datetime.timezone.utc),
    }),
  ])
# ---
# name: test_zone_set_temperature[default]
  list([
    dict({
      'until': datetime.datetime(2024, 7, 10, 21, 10, tzinfo=datetime.timezone.utc),
    }),
  ])
# ---
# name: test_zone_set_temperature[h032585]
  list([
    dict({
      'until': datetime.datetime(2024, 7, 10, 21, 10, tzinfo=datetime.timezone.utc),
    }),
  ])
# ---
# name: test_zone_set_temperature[h099625]
  list([
    dict({
      'until': datetime.datetime(2024, 7, 10, 19, 10, tzinfo=datetime.timezone.utc),
    }),
  ])
# ---
# name: test_zone_set_temperature[minimal]
  list([
    dict({
      'until': datetime.datetime(2024, 7, 10, 21, 10, tzinfo=datetime.timezone.utc),
    }),
  ])
# ---
# name: test_zone_set_temperature[sys_004]
  list([
    dict({
      'until': None,
    }),
  ])
# ---
# name: test_zone_turn_off[default]
  list([
    tuple(
      5.0,
    ),
  ])
# ---
# name: test_zone_turn_off[h032585]
  list([
    tuple(
      4.5,
    ),
  ])
# ---
# name: test_zone_turn_off[h099625]
  list([
    tuple(
      5.0,
    ),
  ])
# ---
# name: test_zone_turn_off[minimal]
  list([
    tuple(
      5.0,
    ),
  ])
# ---
# name: test_zone_turn_off[sys_004]
  list([
    tuple(
      5.0,
    ),
  ])
# ---
